File: readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string s = get_string("Text: ");
    int words, sentences, letters;
    words = 0;
    sentences = 0;
    letters = 0;

    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == '.' || s[i] == '?' || s[i] == '!')
        {
            sentences ++;
        }
        if ((i != n - 1 && s[i] == ' ' && s[i + 1] != ' ') || (i == 0  && s[i] != ' '))
        {
            words ++;
        }
        if (isalpha(s[i]))
        {
            letters ++;
        }
      
    }
    float L = ((float)letters / (float) words) * 100;

    float S = ((float)sentences / (float) words) * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    
        printf("Grade %i\n", index);
        }
    

File: pset6/readability/readability.py
from cs50 import get_string
letter = 0
word = 1
sentence = 0

text = get_string("Text: ")
r = len(text)

for i in range(r): 
    if text[i].isalnum() or text[i].isupper() or text[i].islower() == True: 
        letter = letter+1
for i in range(r):
    if text[i].isspace() == True:
        word = word+1
for i in range(r):
    if text[i]=="?" or text[i]=="." or text[i]=="!":
        sentence = sentence+1
grade = round (0.0588 * ((100 * letter) / word) - 0.296 * ((100 * sentence) / word) - 15.8)
if grade < 1:
    print("Before Grade 1")
elif grade < 16:
    print(f"Grade {grade}")
else:
    print("Grade 16+");